kong/llm/plugin/shared-filters/parse-json-response.lua

```diff
@@ -27,22 +27,6 @@ function _M:run(_)
 
   set_global_ctx("response_body", response_body)
 
-  local t, err
-  if response_body then
-    t, err = cjson.decode(response_body)
-    if err then
-      kong.log.warn("failed to decode response body for usage introspection: ", err)
-    end
-
-    if t and t.usage and t.usage.prompt_tokens then
-      ai_plugin_o11y.metrics_set("llm_prompt_tokens_count", t.usage.prompt_tokens)
-    end
-
-    if t and t.usage and t.usage.completion_tokens then
-      ai_plugin_o11y.metrics_set("llm_completion_tokens_count", t.usage.completion_tokens)
-    end
-  end
-
```

Comment:
this is still needed, when normalize-json-response is not enabled on a request (for example, using semantic-cache without ai proxy)

Reply:
@oowl What do you recommend?

Just duplicate the code for now "QAD" with a `if not (namespace-ai-proxy) then run_the_code end` ?

Reply:
it's fine to run this code twice, the later one will just update/overwrite the previous.
> metadata extraction itself should be move to its own filter too...
Ideally yes, but we are not in the real filter pipeline but still in Kong's plugin iterator.

Right now, move this to llm/shared to a utility function and to be called by both filters will be fine.

Reply:
So ideally:
for ai-proxy
`parse-json-response | get metadata | normalize-json-response | get metadata`
for ai-semantic-cache etc, without ai-proxy
`parse-json-response | get metadata`

Even if we do this, in real world, the plugin iterator still execute all "filters" for one plugin before executing the other, so if you have ai-proxy + ai-semantic-cache it would actually be:

`parse-json-response | get metadata | parse-json-response (skipped) | get metadata | normalize-json-response | get metadata`

so I think it's fine to just have get metadata part of the `parse` and `normalize` so we would have
`parse-json-response| parse-json-response (skipped) | normalize-json-response `

Reply:
I'll add it back and just double check the tests!

Reply:
Oh you already did it! Okay am very confused now.